source/localizable/tutorial/subroutes.md

```diff
@@ -0,0 +1,398 @@
+export default function() {
+  this.get('/rentals', function(db, request) {
+    let rentals = [
+      {
+        type: 'rentals',
+        id: 1,
+        attributes: {
+          "title": "Grand Old Mansion",
+          "owner": "Veruca Salt",
+          "city": "San Francisco",
+          "type": "Estate",
+          "bedrooms": 15,
+          "image": "https://upload.wikimedia.org/wikipedia/commons/c/cb/Crane_estate_(5).jpg",
+          "slug": "grand-old-mansion",
+          "description": "This grand old mansion sits on over 100 acres of rolling hills and dense redwood forests."
+        }
+      },
+      {
+        type: 'rentals',
+        "id": 2,
+        attributes: {
+          "title": "Urban Living",
+          "owner": "Mike Teavee",
+          "city": "Seattle",
+          "type": "Condo",
+          "bedrooms": 1,
+          "image": "https://upload.wikimedia.org/wikipedia/commons/0/0e/Alfonso_13_Highrise_Tegucigalpa.jpg",
+          "slug": "urban-living",
+          "description": "A commuters dream. This rental is within walking distance of 2 bus stops and the Metro."
+        }
+      },
+      {
+        type: 'rentals',
+        "id": 3,
+        attributes: {
+          "title": "Downtown Charm",
+          "owner": "Violet Beauregarde",
+          "city": "Portland",
+          "type": "Apartment",
+          "bedrooms": 3,
+          "image": "https://upload.wikimedia.org/wikipedia/commons/f/f7/Wheeldon_Apartment_Building_-_Portland_Oregon.jpg",
+          "slug": "downtown-charm",
+          "description": "Convenience is at your doorstep with this charming downtown rental. Great restaurants and active night life are within a few feet."
+        }
+      }
+    ];
+
+    if (request.queryParams.city !== undefined) {
+      let filteredRentals = rentals.filter(function (i) {
+        return i.attributes.city.toLowerCase().indexOf(request.queryParams.city.toLowerCase()) !== -1;
+      });
+      return { data: filteredRentals };
+    } else if (request.queryParams.slug !== undefined) {
+      const selectedRental = rentals.find(function(rental) {
+       return rental.attributes.slug === request.queryParams.slug;
+      });
+      return { data: selectedRental };
+    } else {
+      return { data: rentals };
+    }
+  });
+}
+```
+
+Let's quickly highlight the change made to the bottom, where we return our rental.
+
+```mirage/config.js
+else if (request.queryParams.slug !== undefined) {
+  const selectedRental = rentals.find(function(rental) {
+   return rental.attributes.slug === request.queryParams.slug;
+  });
+  return { data: selectedRental };
+}
+```
+
+Our `show` sub-route only expects to show one rental.
+This is why we will use `queryRecord` instead of `query` as we expect a single response.
+This means that we cannot return an array (like we do with a normal get response to our Mirage file).
+We search all of our rentals until we find the one that matches the slug passed in our `queryParams`.
```

Comment:
Yeah.  I'm guess I'm torn here between just using id instead of introducing the slug concept for the tutorial because its more straightforward.  We can leave it for now and I'll try to garner some additional feedback on it.